### PR TITLE
[Docs] Add glossary

### PR DESCRIPTION
**Pull request by jillnogold**

### CODE REVIEW

File: docs/glossary.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+| V3IO                     | Iguazio real-time data layer, supports several formats including KV, Block, File, Streams, and more                                                                                                    |
+| V3IO shard               | Uniquely identified data sets within a V3IO stream. Similar to a Kafka partition                                                                                                              |
+| V3IO stream              | Streaming mechanism part of Iguazio's V3IO data layer. Similar to a Kafka stream                                                                                                              |
+
+## Standard ML Terms
+| Name                                       | Description          |   
+|--------------------------------------------------|---------------------------------------------------------------------------| 
+| Artifact                 | The input or output of experiments and model training runs that  can be used in many runs across the project.                                                                                  |
```
Comment (yaronha) [inline docs/glossary.md:32]: A versioned output of a data processing or model training jobs, can be used as input for other jobs or pipelines in the project. We have various types of artifacts (file, model, dataset, chart, etc.) which incorporate useful metadata.

File: docs/glossary.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+| DAG                      | Directed acyclic graph                                                                                                                                                                        |
```
Comment (yaronha) [inline docs/glossary.md:33]: Directed acyclic graph, used to describe workflows/pipelines.

File: docs/glossary.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+| Feature engineering      | Apply domain knowledge and statistical techniques to raw data to extract more information out of data and improve performance of machine learning models                                      |
+| EDA                      | Exploratory data analysis. Used by data scientists to understand dataset via cleaning, visualization, and statistical tests                                                                   |
+| ML pipeline              | Pipeline of operations for machine learning. It can include loading data, feature engineering, feature selection, model training, hyperparameter tuning, model validation, and model deployment |
+| Feature                  | Raw characteristic or numerical value of a dataset                                                                                                                                            |
```
Comment (yaronha) [inline docs/glossary.md:37]: Data field/vector definitions and metadata (name, type, stats, etc.), a dataset is a collection of features.

File: docs/glossary.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+| KubeFlow pipeline        | Platform for building and deploying portable, scalable machine learning (ML) workflows based on Docker containers                                                                             |
+| PyTorch                  | An open source machine learning framework based on the Torch library, used for applications such as computer vision and natural language processing                                                 |
+| Sklearn                  | Open source machine learning Python library. Used for modelling, pipelines, data transformations, feature engineering, and more.                                                              |
+| Spark                    | Open source parallel processing framework for running large-scale data analytics applications across clustered computers. Often used for data engineering, data science, and machine learning |
+| TensorFlow                    | An end-to-end open source platform for machine learning                                                                                                                                        |
```
Comment (yaronha) [inline docs/glossary.md:52]: A Google developed open-source software library for machine learning and deep learning.

**Review by yaronha — APPROVED** [review]

File: docs/glossary.md (hunk 3 above)
Comment (jillnogold) [reply docs/glossary.md]: updated

File: docs/glossary.md (hunk 2 above)
Comment (jillnogold) [reply docs/glossary.md]: updated

File: docs/glossary.md (hunk 1 above)
Comment (jillnogold) [reply docs/glossary.md]: updated

**Pull request closed without merging**